External/FEXCore/Source/Interface/IR/Passes/ConstProp.cpp

```diff
@@ -559,11 +580,22 @@ bool ConstProp::Run(IREmitter *IREmit) {
       uint64_t Constant1{};
       uint64_t Constant2{};
 
-      if (IREmit->IsValueConstant(Op->Header.Args[0], &Constant1) &&
-          IREmit->IsValueConstant(Op->Header.Args[1], &Constant2)) {
+      auto Is1CST = IREmit->IsValueConstant(Op->Header.Args[0], &Constant1);
+      auto Is2CST = IREmit->IsValueConstant(Op->Header.Args[1], &Constant2);
+
+      if (Is1CST && Is2CST) {
         uint64_t NewConstant = Constant1 | Constant2;
         IREmit->ReplaceWithConstant(CodeNode, NewConstant);
         Changed = true;
+      } /* OR with ~0 is ~0 */ else if ((Is1CST && Constant1 == ~0ULL) || (Is2CST && Constant2 == ~0ULL)) {
```

Comment:
These should be checking against a constant generated by `IROp->Size * 8 - 1` So it works for both 32bit and 64bit.
Then replaced with that constant as well.

Reply:
Converting this to a draft PR until I fix this + make sure there aren't any behavior regressions.